presto-main/src/main/java/com/facebook/presto/sql/gen/CompilerUtils.java

```diff
@@ -97,7 +99,15 @@ public static String toJavaIdentifierString(String className)
         Map<String, byte[]> byteCodes = new LinkedHashMap<>();
         for (ClassDefinition classDefinition : classDefinitions) {
             ClassWriter cw = new SmartClassWriter(classInfoLoader);
-            classDefinition.visit(cw);
+            try {
+                classDefinition.visit(cw);
+            }
+            catch (IndexOutOfBoundsException e) {
+                Printer printer = new Textifier();
+                TraceClassVisitor tcv = new TraceClassVisitor(null, printer, new PrintWriter(System.err));
```

Comment:
Or actually, you could just make a new exception to throw and put the message in there